### ISSUE TITLE
when resizing the browser window the following has to happen

### ISSUE THREAD

**Issue by ba001** [blakearchive/archive#474]
in object view, the gallery window needs to resize accordingly to proportionally fit the new browser window size

in search results, a preview window has to resize accordingly to proportionally fit the new browser window size

this is probably related to #471 (only the 'i' window gets resized)

see production for how it should work.

**Comment by ba001** [comment]
@queryluke whenever you're free, would you mind assisting me here? how were you able to get the gallery and everything in it to adjust to a browser resize? i've just spent three hours fiddling with heights, using 'vh'. i've come close to getting it how it is on production, but i keep getting a glitch, a fix of which causes another glitch

**Comment by queryluke** [comment]
I think you're looking for this directive:

https://github.com/blakearchive/archive/blob/master/blakearchive/static/directives/auto-height/directive.js

Which can be used like this:
https://github.com/blakearchive/archive/blob/master/blakearchive/static/directives/object-viewer/objectViewer.html#L7

The breakpoint variable means the script won't fire below that pixel width. The adjust means you want the height of the element to be X pixels less than the height of the window. In that example, if the window is 1070 vertical pixels, the height of the element will be 800px.

The auto-height (and auto-width) directive is an "angularized" version of the jQuery scripts that dev-hammer provided. I remember trying to accomplish their look & feel using only css, but the way their css and divs were structured, it wasn't possible. So it was easier to turn their jquery into angular directives.

Their jquery code still survives (commented out) in this file:
https://github.com/blakearchive/archive/blob/master/blakearchive/static/js/scripts.js#L88

**Comment by ba001** [comment]
thanks. so i should reduce the breakpoint? not sure i understand. with those numbers it was working before the refactoring, but now it's not. i don't think the refactoring did anything to heights and widths.

**Comment by queryluke** [comment]
the breakpoint is for switching between mobile/desktop, so screens smaller than 992px will let the content be it's natural height.

I can't say what broke in the refactoring. My first test would be to make sure the directive is even firing by logging some message on the window resize.

**Comment by ba001** [comment]
@queryluke ok, yeah, it's firing. newHeight is getting changed, but the new height doesn't actually get rendered on the screen

**Comment by ba001** [comment]
@queryluke autoHeight directive is now broken out of main.js. i think what's happening is that 'element' is not getting passed properly, so all elements with the auto-height directive don't actually get updated on a resize event. the directive is firing, but element.height(newHeight) in directive.js under /auto-height is not actually doing anything

**Comment by ba001** [comment]
looks like div class="reading-wrapper" gets adjusted correctly, too. so that and the 'i' window height, but nothing else using the auto-height directive

**Comment by ba001** [comment]
@nathan-rice could this be a scoping issue that arose from the refactoring? for some reason the element heights aren't getting reset. the newHeight variable is changing, but the styles for the elements using the directive aren't. i'm stumped. this is the last bug on the refactored site to fix so that we can publish a whole bunch of stuff sitting in the queue

**Comment by nathan-rice** [comment]
I just pushed another refactor pass (though the fact that data is a linked repo might have messed up the commit log), I didn't actively intend to fix this bug but resizing behavior appeared the same on london and my local build, so I think it resolved itself in the process of code cleanup.

For future reference, when creating directives that primarily work through a link function, do not create new isolate scopes indiscriminately.  This causes lots of problems.  Instead, get attributes from the linked element using scope.$eval(attrs.attribute_name) - see the updated auto-width/height directives for guidance.

**Comment by ba001** [comment]
great, thanks! fixed

**Comment by ba001** [comment]
@nathan-rice hmm, when i run npm run-script build-watch, i get:

npm ERR! missing script: build-watch

**Comment by ba001** [comment]
not sure why the auto-height isn't working for the containers (columns) in the search preview box. open one, shrink the height of the browser window, and notice that the selected object gets cut off by the bottom of the preview box. it should shrink, along with the containers for the other columns

**Comment by ba001** [comment]
@nathan-rice looks like upon deploy the containers have class=ng-isolate-scope, but they don't in the code

**Comment by ba001** [comment]
i see that the containers create isolate scopes, but they don't work through a link function. should these be converted somehow as well?

**Comment by nathan-rice** [comment]
This is the result of multiple nested auto-height directives that no longer being executed in the proper order due to angular's dom shuffling as a result of splitting search results into directives.

Honestly, this isn't something that should be handled with javascript.  Auto-height should be scrapped, and the HTML/CSS for the relevant part of the sight should be redone to use flexbox, which (supposedly) handles this problem.

**Comment by ba001** [comment]
it looks like it does use flexbox (class="flex-992"), but that isn't working either. could we re-consolidate search results directives for now, just so that we can put this out and start publishing again? we're a little behind schedule

**Comment by ba001** [comment]
@nathan-rice almost there. by taking out the auto-height in the top-level div in the preview-box directive, i managed to get the first and third column adjusting with a browser resize but not the second column (the Selected Object) yet

**Comment by nathan-rice** [comment]
Got it.  Turns out there were actually two related problems:

1. Scopes were being created outside the scope hierarchy, so the event broadcast wasn't propagating.  I changed the listeners to listen on $rootScope, and changed the resize directive to emit on the $rootScope to solve this.
2. setStyle being a function attached to the scope was causing the function to be overwritten.  Not sure exactly why, scope should have been different for each invocation of the directive, but it wasn't.
